<div align="right">
Tyler Grimes <br>
Jan. 19, 2016
</div>

Homework 1 - Ch. 3 #2, 5, 6

<hr>
> 2.) If $x_0 = 3$ and $x_n$ = ($5x_{n-1}$ + 7) mod 200, find $x_1$,...,$x_{10}$.  

In [7]:
x = 3
print "x_0 = 3"
for i in range(1, 11):
    x = (5 * x + 7) % 200
    print "x_%s = %s" % (i, x)

x_0 = 3
x_1 = 22
x_2 = 117
x_3 = 192
x_4 = 167
x_5 = 42
x_6 = 17
x_7 = 92
x_8 = 67
x_9 = 142
x_10 = 117


<p style="page-break-after: always"></p>


<hr>
> 5.) $\int_{-2}^{2} \exp(x + x^2) dx$ 

##### Derivation:  

Let $Y \sim Unif(0,1)$, and let $u = \frac{x + 2}{4}$. Then,
$x = 4u - 2$ and $dx = 4du$. By substitution,

$$\begin{aligned}
\int_{-2}^{2} \exp(x + x^2) dx &= \int_{0}^{1} \exp((4u-2) + (4u-2)^2) 4 du,\\
&= \int_{0}^{1} 4 \exp(16u^2 -12u + 2) f_Y(u) du, \\
&= E_Y(g(u)),
\end{aligned}$$

where $g(u) = 4 \exp(16u^2 -12u + 2)$.  
  
  
##### Algorithm:
1. Generate y from Unif(0,1).
2. Evaluate $x_i = g(y)$.
3. Repeat 1 and 2 for $n = 10^6$ iterations.
4. Evaluate $\frac{\sum_{i=1}^n x_i}{n}$.  
  
  
##### Simulation:
Note, the same algorithm will be used in #5 and #6 of this assignment and is implemented by the function simulation(), as shown in the following code block. The program is written in Python (v 2.7.11).

In [7]:
import math
import random as ran

def simulation(g, seed, n = 1000000):
    sum = 0
    ran.seed(seed)
    
    for i in xrange(n):
        y = ran.uniform(0, 1)
        xi = g(y)
        sum = sum + xi
        
    return sum/n

In [11]:
def g(x):
    return 4*math.exp(16*(x**2) - 12*x + 2)

%time simulation(g, seed = 5)

CPU times: user 897 ms, sys: 27 ms, total: 924 ms
Wall time: 904 ms


93.41366624385321

The simulation yields the approximation $\int_{-2}^{2} \exp(x + x^2) dx \approx 93.41$  
  
  
##### Analytical result:
There is no closed-form solution to $\int_{-2}^{2} \exp(x + x^2) dx$.





<p style:"page-break-before: always"></p>

<hr>
> 6.) $\int_{0}^{\infty} x(1+x^2)^{-2} dx$ 

##### Derivation:  

Let $Y \sim Unif(0,1)$, and let $u = \frac{1}{x+1}$. Then,
$x = \frac{1-u}{u}$ and $dx = -\frac{1}{u^2} du$. By substitution,

$$\begin{aligned}
\int_{0}^{\infty} x(1+x^2)^{-2} dx &= \int_{1}^{0} \left( \frac{1-u}{u} \right) \left(1+ \left( \frac{1-u}{u} \right)^2 \right)^{-2} \left( -\frac{1}{u^2} \right) du,\\
&= \int_{0}^{1} \left( \frac{1-u}{u^3} \right) \left(\frac{u^2}{u^2} + \left( \frac{1-2u+u^2}{u^2}\right) \right)^{-2} du \\
&= \int_{0}^{1} \left( \frac{1-u}{u^3} \right) \left( \frac{u^4}{(1-2u+2u^2)^2} \right) du \\
&= \int_{0}^{1} \left( \frac{(1-u)u}{(1-2u+2u^2)^2} \right) du \\
&= \int_{0}^{1} \left( \frac{(1-u)u}{(1-2u+2u^2)^2} \right) f_Y(u) du \\
&= E_Y(g(u)),
\end{aligned}$$

where $g(u) = \left( \frac{(1-u)u}{(1-2u+2u^2)^2}\right)$.  
  
  
##### Algorithm:
1. Generate y from Unif(0,1).
2. Evaluate $x_i = g(y)$.
3. Repeat 1 and 2 for $n = 10^6$ iterations.
4. Evaluate $\frac{\sum_{i=1}^n x_i}{n}$.  
  
  
##### Simulation:

In [10]:
def g(x):
    return ((1 - x)*(x)) / ((1 - 2*x + 2*x*x)**2)

%time simulation(g, seed = 6)

CPU times: user 883 ms, sys: 41.2 ms, total: 924 ms
Wall time: 896 ms


0.500861713250707

The simulation yields the approximation $\int_{0}^{\infty} x(1+x^2)^{-2} dx \approx 0.5009$  
  
##### Analytical result:

Let $u = 1+x^2$. Then $du = 2xdx$, and
$$\begin{aligned}
\int_{0}^{\infty} x(1+x^2)^{-2} dx &= \int_{1}^{\infty} u^{-2} \frac{du}{2} \\
&= \frac{1}{2} \left[ -u^{-1} \right]_1^{\infty} \\
&= \frac{1}{2} \left[ 0-(-1) \right] \\
&= \frac{1}{2}
\end{aligned}$$
  
Hence, our approximation agrees with the exact result to three decimals.